### PR TITLE
Update vcpkg to latest commit to catch Qt 6.5.0

### PR DESCRIPTION
**Pull request by nirvn**

### CODE REVIEW

**Comment by m-kuhn** [comment]
From `boost-container/install-x64-android-rel-out.log`

```
clang++: error: the clang compiler does not support '-march=i686'
clang-linux.compile.c++ /home/runner/builddir/_deps/vcpkg-src/buildtrees/boost-container/x86-android-rel/boost/build/13e7736a74f8ce1f0320ca511e787056/dlmalloc.o

  "/usr/local/lib/android/sdk/ndk/22.1.7171670/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++" -c -x c++ --target=armv7-none-linux-androideabi21 -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security -frtti -fexceptions -fPIC -fstack-protector-strong -Oz -DNDEBUG "--gcc-toolchain=/usr/local/lib/android/sdk/ndk/22.1.7171670/toolchains/llvm/prebuilt/linux-x86_64" "--sysroot=/usr/local/lib/android/sdk/ndk/22.1.7171670/toolchains/llvm/prebuilt/linux-x86_64/sysroot" -m32 -O3 -Wall -g -Wno-inline -march=i686  -DBOOST_ALL_NO_LIB=1 -DBOOST_CONTAINER_STATIC_LINK=1 -DNDEBUG -I"../include" -I"/home/runner/builddir/vcpkg_installed/x86-android/include"   -o "/home/runner/builddir/_deps/vcpkg-src/buildtrees/boost-container/x86-android-rel/boost/build/13e7736a74f8ce1f0320ca511e787056/dlmalloc.o" "../src/dlmalloc.cpp"
```

I don't know where `--target=armv7-none-linux-androideabi21` comes from in here, we are building for x86

**Comment by nirvn** [comment]
@m-kuhn , cache: true ? :)

**Comment by nirvn** [comment]
(actually no, ignore that, boost was failing when I tried to apply it in system_qt6 branch)

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -171,6 +171,10 @@ jobs:
           curl -sL https://sentry.io/get-cli/ | bash
           sentry-cli upload-dif -o opengisch -p qfield ${{ env.CMAKE_BUILD_DIR }}/android-build/build/intermediates/merged_native_libs/release
 
+      - name: tmate
```
Comment (m-kuhn) [inline .github/workflows/android.yml:174]: TODO: remove before merge

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 3e6424423f849fdfefb020fcaa85ffd6913f7a93: [**arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4282-660467f8b2023810313005ff53b944388273c94f-arm64-android.apk)
<details>
<summary>Other architectures</summary>

- [arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4282-660467f8b2023810313005ff53b944388273c94f-arm-android.apk)
- [x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4282-660467f8b2023810313005ff53b944388273c94f-x64-android.apk)
- [x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-4282-660467f8b2023810313005ff53b944388273c94f-x86-android.apk)
- [**all access arm64-android**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4282-660467f8b2023810313005ff53b944388273c94f-arm64-android.apk)
- [all access arm-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4282-660467f8b2023810313005ff53b944388273c94f-arm-android.apk)
- [all access x64-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4282-660467f8b2023810313005ff53b944388273c94f-x64-android.apk)
- [all access x86-android](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield_all_access-dev-4282-660467f8b2023810313005ff53b944388273c94f-x86-android.apk)
</details>

File: .qt6/vcpkg.json [hunk 2]
```diff
@@ -38,11 +38,6 @@
       "default-features": false
     },
     "nu-book-zxing-cpp",
-    "spix-qt6",
-    {
-      "name": "sentry-native",
```
Comment (m-kuhn) [inline .qt6/vcpkg.json:43]: Sentry also?

Comment (nirvn) [reply .qt6/vcpkg.json:43]: @m-kuhn , oups -- typo, but actually it had to be disabled in system_qt6 too, so maybe let's no re-trigger a build here and try to re-enable it in system_qt6 (or a separate PR altogether after the main qt6 work is done)?

Comment (nirvn) [reply .qt6/vcpkg.json:43]: (i.e. I suspect we'll need to update sentry to 0.6.X, which will be its own interesting struggle :wink:)

**Pull request merged**